futures-test/src/io/interleave_pending.rs

```diff
@@ -0,0 +1,161 @@
+use futures_io::{self as io, AsyncBufRead, AsyncRead, AsyncWrite};
+use pin_utils::{unsafe_pinned, unsafe_unpinned};
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// I/O wrapper for interleaving [`Poll::Pending`] in calls to read or write.
+///
+/// See the [`interleave_pending`] and [`interleave_pending_write`] methods.
+///
+/// [`interleave_pending`]: super::AsyncReadTestExt::interleave_pending
+/// [`interleave_pending_write`]: super::AsyncWriteTestExt::interleave_pending_write
+#[derive(Debug)]
+pub struct InterleavePending<Io> {
```

Comment:
Maybe a silly question: can we also use this for `Future`s / `Stream`s / `Sink`s?

Reply:
Sure, want me to that in this pr?

Reply:
That would be great, thanks! :)